nemo/collections/asr/models/ssl_models.py

```diff
@@ -589,3 +606,406 @@ def multi_validation_epoch_end(self, outputs, dataloader_idx: int = 0):
+            )
+        if not has_processed_noisy_input_signal:
+            processed_noisy_input_signal, processed_noisy_input_signal_length = self.preprocessor(
+                input_signal=noisy_input_signal,
+                length=noisy_input_signal_length,
+            )
+
+        if self.pre_encoder is not None:
+            # mask after convolutional sub-sampling
+            feats, _ = self.pre_encoder.pre_encode(x=processed_signal, lengths=processed_signal_length)
+            _, tokens = self.quantizer(input_signal=feats.transpose(1, 2))
+
+            self.pre_encoder.set_masking_enabled(apply_mask=apply_mask)
+            encoded, encoded_len = self.encoder(
+                audio_signal=processed_noisy_input_signal, length=processed_noisy_input_signal_length
+            )
+            masks = self.pre_encoder.get_current_mask()
+        else:
+            _, tokens = self.quantizer(input_signal=processed_signal)
+            if apply_mask:
+                masked_signal, masks = self.mask_processor(
+                    input_feats=processed_noisy_input_signal, input_lengths=processed_noisy_input_signal_length
+                )
+            else:
+                masked_signal = processed_noisy_input_signal
+                masks = torch.zeros_like(processed_noisy_input_signal)
+            encoded, encoded_len = self.encoder(audio_signal=masked_signal, length=processed_noisy_input_signal_length)
+
+        log_probs = self.decoder(encoder_output=encoded)
+
+        return log_probs, encoded_len, masks, tokens
+
+    def training_step(self, batch: ssl_dataset.AudioNoiseBatch, batch_idx: int):
+        log_probs, encoded_len, masks, tokens = self.forward(
+            input_signal=batch.audio,
+            input_signal_length=batch.audio_len,
+            noise_signal=batch.noise,
+            noise_signal_length=batch.noise_len,
+            noisy_input_signal=batch.noisy_audio,
+            noisy_input_signal_length=batch.noisy_audio_len,
+            apply_mask=True,
+        )
+
+        loss_value = self.loss(masks=masks, decoder_outputs=log_probs, targets=tokens, decoder_lengths=encoded_len)
+
+        tensorboard_logs = {
+            'learning_rate': self._optimizer.param_groups[0]['lr'],
+            'global_step': self.trainer.global_step,
+            'train_loss': loss_value,
+        }
+
+        return {'loss': loss_value, 'log': tensorboard_logs}
+
+    def inference_pass(
+        self, batch: ssl_dataset.AudioNoiseBatch, batch_idx: int, dataloader_idx: int = 0, mode: str = 'val'
+    ):
+        log_probs, encoded_len, masks, tokens = self.forward(
+            input_signal=batch.audio,
+            input_signal_length=batch.audio_len,
+            noise_signal=batch.noise,
+            noise_signal_length=batch.noise_len,
+            noisy_input_signal=batch.noisy_audio,
+            noisy_input_signal_length=batch.noisy_audio_len,
+            apply_mask=True,
```

Comment:
⚠️ masking is enabled during inference pass here.

Reply:
it's intended for validation. during actual usage of ssl model, only the forward() function is used, which by default doesn't apply masking.